### ISSUE TITLE
set-env-java-home not persistent and executed on every chef client run - is this needed?

### ISSUE THREAD

**Issue by richardowen** [sous-chefs/java#246]
I'm referring to these lines:

https://github.com/agileorbit-cookbooks/java/blob/master/recipes/set_java_home.rb#L19-L24

**Comment by richardowen** [comment]
The env variable change isn't persisted, it's just within the chef-client process. This means it executes on every chef client run and therefore reports that resources are being updated (not great for looking at what your nodes are executing).

Does this actually need to be there, since it isn't currently working?

**Comment by majormoses** [comment]
This is an old issue I assume you are talking about these lines: https://github.com/sous-chefs/java/blob/v2.2.1/recipes/set_java_home.rb#L28-L36 honestly reading that code I think the problem is that we should only be running:
```
file.search_file_replace_line(/^JAVA_HOME=/, "JAVA_HOME=#{node['java']['java_home']}")
```
if this actually resulted in a change:
```
file.insert_line_if_no_match(/^JAVA_HOME=/, "JAVA_HOME=#{node['java']['java_home']}")
```

**Comment by majormoses** [comment]
Actually I read that backwards I think its legit and this is probably an old issue that does not apply anymore.

**Comment by majormoses** [comment]
fixed via #394
